nuxeo-drive-core/src/main/java/org/nuxeo/drive/listener/NuxeoDriveGroupUpdateListener.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * @since 9.2
+ */
+public class NuxeoDriveGroupUpdateListener implements PostCommitFilteringEventListener {
+
+    protected List<String> acceptedEvents = Arrays.asList(UserManagerImpl.GROUPCREATED_EVENT_ID,
+            UserManagerImpl.GROUPDELETED_EVENT_ID, UserManagerImpl.GROUPMODIFIED_EVENT_ID);
+
+    @Override
+    public boolean acceptEvent(Event event) {
+        return event.getContext() != null
+                & UserManagerImpl.USER_GROUP_CATEGORY.equals(
+                        event.getContext().getProperty(DocumentEventContext.CATEGORY_PROPERTY_KEY))
+                && acceptedEvents.contains(event.getName());
+    }
+
+    @Override
+    public void handleEvent(EventBundle events) {
+        for (Event event : events) {
+            EventContext context = event.getContext();
+            if (context == null) {
+                return;
+            }
+            String groupName = (String) context.getProperty(UserManagerImpl.ID_PROPERTY_KEY);
+            if (StringUtils.isBlank(groupName)) {
+                return;
+            }
+            List<String> groupNames = getAllGroupNames(groupName, context);
+            handleUpdatedGroups(groupNames);
+        }
+    }
+
+    /**
+     * Returns a list containing the names of the given group and all its ancestor groups.
+     */
+    @SuppressWarnings("unchecked")
+    protected List<String> getAllGroupNames(String groupName, EventContext context) {
+        List<String> groupNames = new ArrayList<String>();
+        groupNames.add(groupName);
+        // Get ancestor groups from the event context or compute them if not provided
+        List<String> ancestorGroups = (List<String>) context.getProperty(UserManagerImpl.ANCESTOR_GROUPS_PROPERTY_KEY);
+        if (ancestorGroups != null) {
+            groupNames.addAll(ancestorGroups);
+        } else {
+            groupNames.addAll(Framework.getService(UserManager.class).getAncestorGroups(groupName));
+        }
+        return groupNames;
+    }
+
+    protected void handleUpdatedGroups(List<String> groupNames) {
+        RepositoryManager repositoryManager = Framework.getService(RepositoryManager.class);
+        for (String repositoryName : repositoryManager.getRepositoryNames()) {
+            CoreSession session = CoreInstance.openCoreSessionSystem(repositoryName);
+            try {
+                DocumentModelList impactedDocuments = getImpactedDocuments(session, groupNames);
+                impactedDocuments.forEach(doc -> fireGroupUpdatedEvent(session, doc));
+            } finally {
+                session.close();
+            }
+        }
+    }
+
+    /**
+     * Returns the list of documents carrying an ACL impacted by one of the given group names.
+     */
+    protected DocumentModelList getImpactedDocuments(CoreSession session, List<String> groupNames) {
+        String groups = String.join(",",
+                groupNames.stream().map(group -> "'" + group + "'").collect(Collectors.toList()));
```

Comment:
You can use `Collectors.joining(delimiter, prefix, suffix)` to simplify all this.